staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -223,6 +228,40 @@ func detectSystemd() bool {
 	return true
 }
 
+// detectSafeNotMountedBehavior returns true if the umount implementation replies "not mounted"
+// when the specified path is not mounted. When not sure (permission errors, ...), it returns false.
+// When possible, we will trust umount's message and avoid doing our own mount point checks.
+// More info: https://github.com/util-linux/util-linux/blob/v2.2/mount/umount.c#L179
+func detectSafeNotMountedBehavior() bool {
+	return detectSafeNotMountedBehaviorWithExec(utilexec.New())
+}
+
+// detectSafeNotMountedBehaviorWithExec is for testing with FakeExec.
+func detectSafeNotMountedBehaviorWithExec(exec utilexec.Interface) bool {
+	if _, err := exec.LookPath("umount"); err != nil {
+		klog.V(2).Infof("Failed to locate umount executable to detect safe 'not mounted' behavior")
```

Comment:
This is a library and we should not have logging; can we only have debug logs as there a lot of cases where it might be useful?

Reply:
maybe make it level 4?

Also in the below func (mounter *Mounter) Unmount(target string), it does not try to lookPath for umount, so is it needed here?

Reply:
I've changed these to V(4) and removed the LookPath.